ckan/controllers/user.py

```diff
@@ -162,10 +166,90 @@ def register(self, data=None, errors=None, error_summary=None):
 
         return self.new(data, errors, error_summary)
 
+    def _generate_internal_logs(self, log_message):
+        '''
+        This helper function is called if sending a new user registration email fails.
+        '''
+
+        logging.warning("There was an error sending the email. Writing to log file.")
+
+        log_path = os.environ['REGISTRATION_ERROR_LOG_FILE_PATH']
+
+        try:
+            with open(log_path + '/' + os.environ['REGISTRATION_ERROR_LOG_FILE_NAME'], "a") as fp:
+                try:
+                    fp.write(log_message)
+                    fp.write("\n\n")
+                except:
+                    logging.warning("There is an error writing to the log file. Dumping the output to console.")
+                    logging.warning(log_message)
+        except:
+            logging.warning("There is an error writing to the log file. Dumping the output to console.")
+            logging.warning(log_message)
+
+        logging.warning("Written to log file.")
+
+    def email_new_user_details_using_mailgun(self, request_data):
+        request_params = dict(request_data)
+
+        email_body = "There is a new user registration request. \
+        \nThe user's details are as follows: \
+        \n\
+        \nUsername: {username}\
+        \nName: {name} \
+        \nEmail: {email} \
+        \nReason for Request: {reason_for_request} \
+        \nProject of Interest: {project_of_interest} ".format(\
+        username=request_params['name'], \
+        name=request_params['fullname'], \
+        email=request_params['email'], \
+        reason_for_request=request_params['request_reason'], \
+        project_of_interest=request_params['project_of_interest'])
+
+        MAILGUN_ENVIRON_VARS = ['MAILGUN_API_KEY', 'MAILGUN_API_DOMAIN', 'MAILGUN_SENDER_EMAIL', 'MAILGUN_RECEIVER_EMAIL']
+
+        MAILGUN_VARS = dict ((t, os.environ.get(t)) for t in MAILGUN_ENVIRON_VARS)
+
+        for key, value in MAILGUN_VARS.iteritems():
```

Comment:
very minor, but this will log N times where N is the number of unset environment variables. better to just do `if None in MAILGUN_VARS.values()` and print out an error. it wouldn't take whoever it is long to figure out which env variable is missing.

Reply:
Fixed.